src/backend/TrafficCourts/Workflow.Service/Consumers/GetDisputeConsumer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using MassTransit;
+using TrafficCourts.Messaging.MessageContracts;
+using TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0;
+using TrafficCourts.Workflow.Service.Services;
+
+namespace TrafficCourts.Workflow.Service.Consumers
+{
+    /// <summary>
+    ///     Consumer for GetDispute message, which will return dispute
+    /// </summary>
+    public class GetDisputeConsumer : IConsumer<GetDisputeRequest>
+    {
+        private readonly ILogger<GetDisputeConsumer> _logger;
+        private readonly IOracleDataApiService _oracleDataApiService;
+
+        public GetDisputeConsumer(ILogger<GetDisputeConsumer> logger, IOracleDataApiService oracleDataApiService)
+        {
+            _logger = logger;
+            _oracleDataApiService = oracleDataApiService ?? throw new ArgumentNullException(nameof(oracleDataApiService));
+        }
+
+        /// <summary>
+        /// Consumer for getting the dispute
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async Task Consume(ConsumeContext<GetDisputeRequest> context)
+        {
+            using var scope = _logger.BeginConsumeScope(context);
+            Dispute? searchResult;
+            try
+            {
+                searchResult = await _oracleDataApiService.GetDisputeByNoticeOfDisputeGuidAsync(context.Message.NoticeOfDisputeGuid, context.CancellationToken);
+                if (searchResult is null)
+                {
+                    throw new Exception("Dispute not found");

```

Comment:
Do not throw generic exception. Also, do not throw exception like this from a message consumer. If the consumer throws, the message will be requeued and processed again potentially. Additionally, the MassTransit reports metrics regarding the `Number of message consume faults` which means a consumer threw an exception. Having these false positives diminishes the value of metrics.